client/tests/integration/queries/role.rs

```diff
@@ -125,8 +124,15 @@ fn find_roles_by_account_id() -> Result<()> {
         .cloned()
         .map(|role_id| {
             RegisterBox::new(
-                Role::new(role_id)
-                    .add_permission(CanSetKeyValueInUserMetadata::new(alice_id.clone())),
+                Role::new(role_id).add_permission(
+                    PermissionToken::new(
+                        "can_set_key_value_in_user_account".parse().expect("Valid"),
+                    )
+                    .with_params([(
+                        "account_id".parse().expect("Valid"),
+                        alice_id.clone().into(),
+                    )]),
```

Comment:
so we're back to untyped permission tokens?

Reply:
also, does it have to be a tuple `account_id: <account_id>`? They can be treated as positional arguments instead of mapping?

Reply:
> so we're back to untyped permission tokens?

Yes, that's one of the things I don't like. It will be cool to have a way to get them from validators.

From the other side: we need this only for tests. And current test are deprecated in favor of future Python tests. I recommend to wait for #3230 . After that I hope we can find a way to get all permisison tokens defined in validator as some kind of validator's metadata. And then use their schema in Python

Reply:
> also, does it have to be a tuple account_id: <account_id>? They can be treated as positional arguments instead of mapping?

So your question is `Why do we need named parameters, when we can have just wrappers?`, right? Well, I believe, for readability inside transactions and consitency with cases when we have more than one parameter (although we don't have these cases for now)

Reply:
```suggestion
                    .with_params([alice_id.clone().into()]),
```

My question is why is it not done like this? Why do we store a map instead of a list of parameters

Reply:
1. This was done in initial permission tokens
2. It's easy to read. For example in `genesis.json`
3. This question actually is also applicable to instruction parameters

I like it mostly for readability.

Do you think we should consider removing map and use list instead? I think, then we should also consider this for instructions

Reply:
I lean towards pairing the declaration of the new permission validator with its tokens. Without each other, they're meaningless and both registrations should happen simultaneously to prevent breakage. The `metadata` approach seems like  a logical choice, but I'd err on the side of caution and make it a separate hashmap. 

This should let you query for the permission token definitions just like right now, but more compactly, and hopefully more consistently also. 

As for typing. IMO, until we have a functional DSL prototype it's too early to commit one way or another. If we decide that type inference is not possible, having duck-typing will help. If we find that it's easy to implement type inference without too much extra work then strictly typed tokens would be good.

Reply:
> My question is why is it not done like this? Why do we store a map instead of a list of parameters?

We could re-evaluate this in the context of the DSL. Named parameters are better when you have duck-typing. Positional arguments are preferable when you can query what something's type is and that type hints where things should go. 

To me, regardless of which way we go, the current approach is the best. Going with positional arguments makes it harder to interpret the objects correctly. Example: when @arndey and I had to udpate the SDKs, having access to the field names helped a lot. I'd stick with this approach, unless we already fixed the API.

Reply:
I'm also for leaving this as map.